sdktests/custom_matchers_general.go

```diff
@@ -21,6 +22,23 @@ import (
 // The functions in this file are for convenient use of the matchers API with complex
 // types. For more information, see matchers.Transform.
 
+// QueryParameters returns a MatcherTransform which parses a string representing a URL's
+// RawQuery field into a map from parameter key to parameter value. If there are multiple values
+// for a key, the first is used.
```

Comment:
Maybe multiple values could be an error? It feels problematic for testing if you make a  URL with something like `filter=potato&filter=bacon` that if the desired filter was `potato` that it would pass?

Reply:
I missed that when I refactored to use a MatcherTransform here, will fix.